client/blocks/importer/wordpress/upgrade-plan/test/index.tsx

```diff
@@ -0,0 +1,206 @@
+/**
+ * @jest-environment jsdom
+ */
+import { PLAN_MIGRATION_TRIAL_MONTHLY } from '@automattic/calypso-products';
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import nock from 'nock';
+import React, { type ComponentPropsWithoutRef } from 'react';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import { renderWithProvider } from 'calypso/test-helpers/testing-library';
+import { UpgradePlan } from '../index';
+
+// Stub out UpgradePlanDetails because it has much more complex dependencies, and only provides a wrapper around the content from this component.
+jest.mock( '../upgrade-plan-details', () => ( {
+	__esModule: true,
+	default: ( { children } ) => <div>{ children }</div>,
+} ) );
+
+// This is effectively a custom passthrough as jest.spyOn() doesn't seem to work well
+// with the recordTracksEvent import.
+jest.mock( 'calypso/state/analytics/actions', () => {
```

Comment:
Do we have any specially reason in using the action instead of the of the regular `@automattic/calypso-analytics`

it is way more simple to use and test. 

Here an example of test using https://github.com/Automattic/wp-calypso/blob/bc62062eb018b0b0e4fa94e540bf2da1d294464f/packages/launchpad/src/use-tracking/test/index.tsx#L7-L8

Reply:
I mostly didn't want to change the underlying code unnecessarily. But I think it's worth using the library more directly, so I'll tackle that in this PR.
f1e583f7740fba2d6965d57c8a48074c6dc32277